### ISSUE TITLE
Accuracy problem

### ISSUE THREAD

**Issue by ttthappy** [yysijie/st-gcn#120]
@yysijie Hello, thank you for your kindly sharing the st-gcn code. I want to train the st-gcn model in NTU RGB+D dataset. I use the old version code with default parameters. But, after 80 epoch's training I got poor result.
Then I modify the learning rate according to your paper, but the accuracy is still poor（Top1: 1.67%, Top5: 8.27%）.
As a new student in this field, I was puzzled if I missed some details. Hope your reply. Thanks a lot！
  
[ Sat Dec 22 18:59:51 2018 ] Training epoch: 80
[ Sat Dec 22 19:00:03 2018 ] 	Batch(0/589) done. Loss: 4.1821  lr:0.001000
[ Sat Dec 22 19:03:31 2018 ] 	Batch(100/589) done. Loss: 4.1529  lr:0.001000
[ Sat Dec 22 19:06:57 2018 ] 	Batch(200/589) done. Loss: 4.1635  lr:0.001000
[ Sat Dec 22 19:10:27 2018 ] 	Batch(300/589) done. Loss: 4.1190  lr:0.001000
[ Sat Dec 22 19:13:55 2018 ] 	Batch(400/589) done. Loss: 4.1429  lr:0.001000
[ Sat Dec 22 19:17:22 2018 ] 	Batch(500/589) done. Loss: 4.1611  lr:0.001000
[ Sat Dec 22 19:20:24 2018 ] 	Mean training loss: 4.1409.
[ Sat Dec 22 19:20:24 2018 ] 	Time consumption: [Data]01%, [Network]99%
[ Sat Dec 22 19:20:24 2018 ] Eval epoch: 80
[ Sat Dec 22 19:24:07 2018 ] 	Mean test loss of 296 batches: 4.106204550008516.
[ Sat Dec 22 19:24:08 2018 ] 	Top1: 1.26%
[ Sat Dec 22 19:24:08 2018 ] 	Top5: 9.44%

modify the learning rate according to your paper:

[ Tue Dec 25 06:20:27 2018 ] Training epoch: 20
[ Tue Dec 25 06:20:48 2018 ] 	Batch(0/589) done. Loss: 4.5612  lr:0.001000
[ Tue Dec 25 06:25:36 2018 ] 	Batch(100/589) done. Loss: 4.8982  lr:0.001000
[ Tue Dec 25 06:30:21 2018 ] 	Batch(200/589) done. Loss: 4.5673  lr:0.001000
[ Tue Dec 25 06:35:04 2018 ] 	Batch(300/589) done. Loss: 4.6968  lr:0.001000
[ Tue Dec 25 06:39:46 2018 ] 	Batch(400/589) done. Loss: 4.5363  lr:0.001000
[ Tue Dec 25 06:44:30 2018 ] 	Batch(500/589) done. Loss: 4.4839  lr:0.001000
[ Tue Dec 25 06:48:41 2018 ] 	Mean training loss: 4.6989.
[ Tue Dec 25 06:48:41 2018 ] 	Time consumption: [Data]01%, [Network]99%
[ Tue Dec 25 06:48:41 2018 ] Eval epoch: 20
[ Tue Dec 25 06:53:54 2018 ] 	Mean test loss of 296 batches: 4.28557998747439.
[ Tue Dec 25 06:53:54 2018 ] 	Top1: 1.52%
[ Tue Dec 25 06:53:55 2018 ] 	Top5: 8.95%

**Comment by ttthappy** [comment]
感谢您的热心解答！您用的是v0.1.0版本程序训练的吗？我用的是v0.1.0在文章中的nturgb+d数据集（60类）上训练的，没有改任何参数，但是识别率不正常。您方便给一个qq号或邮箱吗？想具体请教一下。非常感谢！

**Comment by Zakeiswo** [comment]
@ttthappy 我用的不是v0.1.0，用的新的版本，我刚刚看了看v0.1.0版本默认的num_class参数并没写错。那我就不知道是什么问题了，sorry
联系方式就不留了，我也是新手，大家一起在这里讨论吧，哈哈

**Comment by ttthappy** [comment]
@Zakeiswo 感谢您的热心解答！

**Comment by hd120105** [comment]
> @ttthappy 我用的不是v0.1.0，用的新的版本，我刚刚看了看v0.1.0版本默认的num_class参数并没写错。那我就不知道是什么问题了，sorry
> 联系方式就不留了，我也是新手，大家一起在这里讨论吧，哈哈



> I think you can try to add "weights: xxxxxxx.pt" in train.yaml to train more times with the last weights.

您好，冒昧打扰您，我在一个新的数据集上进行训练，使用原论文提供的预训练模型，但是出现错误：size mismatch for fcn.weight:copying a param of torch.size([100,256,1,1]) from checkpoint,where the shape is torch.Size([400,256,1,1]) in current model
我现在的训练及有100类，使用论文中kinetics-skeleton有400类，您能告诉我修改哪里能解决这个问题吗

**Comment by pingqi** [comment]
@ttthappy @Zakeiswo @hd120105 @cclauss @yjxiong ，请问，各位，能把作者的相关模型下载下来吗？我从https://s3-us-west-1.amazonaws.com/yysijie-data/public/st-gcn/models/这里下不下来，能否分享我下呢？如果不方便的话，可以把链接发到我的邮箱（pingqi_123@163.com）。谢谢各位了。

**Comment by yysijie** [comment]
@ttthappy, hi, can you provide the command you used for training?

**Comment by yysijie** [comment]
@pingqi, try to download models [here](https://pan.baidu.com/s/1dwKG2TLvG-R1qeIiE4MjeA#list/path=%2FShare%2FAAAI18%2Fst-gcn%2Fmodels&parentPath=%2FShare).
And put them under `./models`.

**Comment by ttthappy** [comment]
@yysijie  Happy New Year!Thank you for your kindly sharing the st-gcn code. I used the v0.1.0 code and the command in the README for training , 
  python main.py --config config/st_gcn/nturgbd-cross-subject/train.yaml
  python main.py --config config/st_gcn/nturgbd-cross-view/train.yaml
Thank you for your reply.

**Comment by ttthappy** [comment]
@yysijie  Hi, thank you for your kindly sharing the st-gcn code. I used the v0.1.0 code and the command in the README for training , 
python main.py --config config/st_gcn/nturgbd-cross-subject/train.yaml
python main.py --config config/st_gcn/nturgbd-cross-view/train.yaml

 The learning rate is set according to your paper or by default:
[ Wed Jan  9 17:33:53 2019 ] Parameters:
{'work_dir': './work_dir/NTU-RGB-D/xsub/ST_GCN', 'config': './config/st_gcn/nturgbd-cross-subject/train.yaml', 'phase': 'train', 'save_score': False, 'seed': 1, 'log_interval': 100, 'save_interval': 10, 'eval_interval': 5, 'print_log': True, 'show_topk': [1, 5], 'feeder': 'st_gcn.feeder.Feeder', 'num_worker': 128, 'train_feeder_args': {'data_path': './data/NTU-RGB-D/xsub/train_data.npy', 'label_path': './data/NTU-RGB-D/xsub/train_label.pkl'}, 'test_feeder_args': {'data_path': './data/NTU-RGB-D/xsub/val_data.npy', 'label_path': './data/NTU-RGB-D/xsub/val_label.pkl'}, 'model': 'st_gcn.net.ST_GCN', 'model_args': {'num_class': 60, 'channel': 3, 'window_size': 300, 'num_point': 25, 'num_person': 2, 'graph': 'st_gcn.graph.NTU_RGB_D', 'graph_args': {'labeling_mode': 'spatial'}, 'mask_learning': True, 'use_data_bn': True}, 'weights': None, 'ignore_weights': [], 'base_lr': 0.1, 'step': [10, 50], 'device': [1, 2, 3], 'optimizer': 'SGD', 'nesterov': True, 'batch_size': 64, 'test_batch_size': 64, 'start_epoch': 0, 'num_epoch': 80, 'weight_decay': 0.0001, 'display_by_category': False, 'display_recall_precision': False}

Thank you for your reply. Happy New Year!

**Comment by pingqi** [comment]
@yysijie Hi, is the log of my training process normal?

**Comment by yysijie** [comment]
@pingqi, do not worry, it's normal. You do not need to download videos. Our codebase only focuses on human skeleton sequence.

**Comment by yysijie** [comment]
@ttthappy, it is strange. 
I will try to replicate the experiments according to your description. 
By the way, how did you to perform data processing?

**Comment by ttthappy** [comment]
@yysijie Thank you for your reply. I performed data processing according to readme,
python tools/ntu_gendata.py --data_path /home/tangjun/dataset/nturgb+d_skeletons

I git clone the code, and git checkout v0.1.0 . Pytorch v0.2.0.
Almost all operations are follow your readme, except manually adding directories data/Kinetics/kinetics-skeleton in data preparation for Kinetics-skeleton.
But after 80 epoch's training the results in NTU RGB+D dataset is strangely poor(Top1: 1.26%). 

Due to the time consumption in training for kinetics-skeleton, I trained 5 epochs:
[ Sat Jan 19 07:12:42 2019 ] Eval epoch: 5
[ Sat Jan 19 07:22:24 2019 ]    Mean test loss of 310 batches: 4.51230739085905.
[ Sat Jan 19 07:22:29 2019 ]    Top1: 10.49%
[ Sat Jan 19 07:22:30 2019 ]    Top5: 26.59%
I'm not sure if this result is right.

Thank you very much!

**Comment by pingqi** [comment]
@yysijie Hi, after 60 epochs,the acc is as follows, while in the paper the acc is 30.7%.So which factor can be changed to improve the acc?
Ps:Does the code can print the acc of a specific action as walking the dog,not only the whole acc?
[02.15.19|15:17:34] Eval epoch: 59
[02.15.19|15:19:21]     mean_loss: 4.264934691467671
[02.15.19|15:19:21]     Top1: 19.28%
[02.15.19|15:19:22]     Top5: 39.37%

**Comment by yysijie** [comment]
@ttthappy, @pingqi,
don't worry, we will release training configuration files and pretrained models on NTU-RGB+D on the latest code base in this week. I think it will deal with most problems you mentioned.

**Comment by ttthappy** [comment]
@yysijie Thank you for your kindly help!

**Comment by yysijie** [comment]
Hi, @ttthappy @Zakeiswo @hd120105 @pingqi,
we provide the training scripts and pretrained models on NTU-RGB+D.
Update the codebase, download models again, then you can perform testing and training.
I think you can achieve the performance we claimed in the paper this time.

**Comment by yysijie** [comment]
Feel free to open another issue if there are other problems.

**Comment by dedoogong** [comment]
Dear @yysijie, I'm trying to improve your model and testing it. 
Could you please upload NTU RGB model on Google drive too? I can't access Baidu drive because it's so hard to make the account for foreigners. 

Thank you!

**Comment by wwwpbai** [comment]
@hd120105 , you should remove the last layer of the pretrained model and revise the number class when you use pretrained model. you can look at [this](https://github.com/Sindreedelange/stgcn/blob/master/torchlight/torchlight/io.py)

**Comment by gravesprite** [comment]
> @yysijie Hello, thank you for your kindly sharing the st-gcn code. I want to train the st-gcn model in NTU RGB+D dataset. I use the old version code with default parameters. But, after 80 epoch's training I got poor result.
> Then I modify the learning rate according to your paper, but the accuracy is still poor（Top1: 1.67%, Top5: 8.27%）.
> As a new student in this field, I was puzzled if I missed some details. Hope your reply. Thanks a lot！
> 
> [ Sat Dec 22 18:59:51 2018 ] Training epoch: 80
> [ Sat Dec 22 19:00:03 2018 ] Batch(0/589) done. Loss: 4.1821 lr:0.001000
> [ Sat Dec 22 19:03:31 2018 ] Batch(100/589) done. Loss: 4.1529 lr:0.001000
> [ Sat Dec 22 19:06:57 2018 ] Batch(200/589) done. Loss: 4.1635 lr:0.001000
> [ Sat Dec 22 19:10:27 2018 ] Batch(300/589) done. Loss: 4.1190 lr:0.001000
> [ Sat Dec 22 19:13:55 2018 ] Batch(400/589) done. Loss: 4.1429 lr:0.001000
> [ Sat Dec 22 19:17:22 2018 ] Batch(500/589) done. Loss: 4.1611 lr:0.001000
> [ Sat Dec 22 19:20:24 2018 ] Mean training loss: 4.1409.
> [ Sat Dec 22 19:20:24 2018 ] Time consumption: [Data]01%, [Network]99%
> [ Sat Dec 22 19:20:24 2018 ] Eval epoch: 80
> [ Sat Dec 22 19:24:07 2018 ] Mean test loss of 296 batches: 4.106204550008516.
> [ Sat Dec 22 19:24:08 2018 ] Top1: 1.26%
> [ Sat Dec 22 19:24:08 2018 ] Top5: 9.44%
> 
> modify the learning rate according to your paper:
> 
> [ Tue Dec 25 06:20:27 2018 ] Training epoch: 20
> [ Tue Dec 25 06:20:48 2018 ] Batch(0/589) done. Loss: 4.5612 lr:0.001000
> [ Tue Dec 25 06:25:36 2018 ] Batch(100/589) done. Loss: 4.8982 lr:0.001000
> [ Tue Dec 25 06:30:21 2018 ] Batch(200/589) done. Loss: 4.5673 lr:0.001000
> [ Tue Dec 25 06:35:04 2018 ] Batch(300/589) done. Loss: 4.6968 lr:0.001000
> [ Tue Dec 25 06:39:46 2018 ] Batch(400/589) done. Loss: 4.5363 lr:0.001000
> [ Tue Dec 25 06:44:30 2018 ] Batch(500/589) done. Loss: 4.4839 lr:0.001000
> [ Tue Dec 25 06:48:41 2018 ] Mean training loss: 4.6989.
> [ Tue Dec 25 06:48:41 2018 ] Time consumption: [Data]01%, [Network]99%
> [ Tue Dec 25 06:48:41 2018 ] Eval epoch: 20
> [ Tue Dec 25 06:53:54 2018 ] Mean test loss of 296 batches: 4.28557998747439.
> [ Tue Dec 25 06:53:54 2018 ] Top1: 1.52%
> [ Tue Dec 25 06:53:55 2018 ] Top5: 8.95%

Hi, have you resolved your problem, recently I follow the train.yaml to train NTU dataset, but got similar result as yours. Could not find a way out. So I really wanna know how you solved your problem, thnaks.

**Comment by 1771203081** [comment]
@gravesprite  Have you solved it? I have this problem, too

**Comment by gravesprite** [comment]
> @gravesprite Have you solved it? I have this problem, too

NO. problem still exist, but a found a way out.
The old version of [st-gcn](https://github.com/yysijie/st-gcn), I have no problem to run with that. I strongly doubt it's because the use of mmcv module. Maybe you should try the old version, it could make you on your way.

**Comment by 1771203081** [comment]
> > @gravesprite Have you solved it? I have this problem, too
> 
> NO. problem still exist, but I found a way out.
> The old version of [st-gcn](https://github.com/yysijie/st-gcn), I have no problem to run with that. I strongly doubt it's because the use of mmcv module. Maybe you should try the old version, it could make you on your way.

thansk you very much

**Comment by liqier** [comment]
> > @ttthappy 我用的不是v0.1.0，用的新的版本，我刚刚看了看v0.1.0版本默认的num_class参数并没写错。那我就不知道是什么问题了，sorry
> > 联系方式就不留了，我也是新手，大家一起在这里讨论吧，哈哈
> 
> > I think you can try to add "weights: xxxxxxx.pt" in train.yaml to train more times with the last weights.
> 
> 您好，冒昧打扰您，我在一个新的数据集上进行训练，使用原论文提供的预训练模型，但是出现错误：size mismatch for fcn.weight:copying a param of torch.size([100,256,1,1]) from checkpoint,where the shape is torch.Size([400,256,1,1]) in current model
> 我现在的训练及有100类，使用论文中kinetics-skeleton有400类，您能告诉我修改哪里能解决这个问题吗
@ttthappy我遇到了和您相同的问题，请问最终您解决了吗？
